internal/declarative/v2/reconciler.go

```diff
@@ -296,6 +294,27 @@ func (r *Reconciler) syncResources(
 	return r.checkTargetReadiness(ctx, clnt, obj, target)
 }
 
+func hasDiff(oldResources []Resource, newResources []Resource) bool {
+	if len(oldResources) != len(newResources) {
+		return true
+	}
+	countMap := map[string]bool{}
+	for _, item := range oldResources {
+		countMap[item.ID()] = true
+	}
+	for _, item := range newResources {
+		if countMap[item.ID()] {
+			countMap[item.ID()] = false
+		}
+	}
+	for _, exists := range countMap {
+		if exists {
+			return true
+		}
+	}
+	return false
```

Comment:
A bit more efficient implementation.
Implicitly assumes that `item.ID()` is unique in both collections, but your code also requires that.
Should it return true if a new resource is detected, but the length of the two collections is the same? I assumed so.

```suggestion
	var diffs = 0
	countMap := map[string]bool{}
	for _, item := range oldResources {
		countMap[item.ID()] = true
		diffs += 1
	}
	for _, item := range newResources {
		if countMap[item.ID()] {
			diffs -= 1
		} else {
			return true //a new resource detected
		}
	}
	
	return diffs != 0
```

Reply:
this solution have some edge case problem, it will return true if `newResources` list contains duplicate resources but the list size is the same as `oldResources`, check the updated [unit test case](https://github.com/kyma-project/lifecycle-manager/pull/730/files#diff-608c0e7ac0ff9d6c47c49efcc9c990d99b696100b56632bb72d961ec37d7e3a5R113-R116) as example

Reply:
I enhanced the parsing logic to remove duplication, please [check](https://github.com/kyma-project/lifecycle-manager/pull/730/files#diff-4a98c02a04a96038d6875b39e833779fc588b08dcf9ec098b46b572d5bd3583eR38-R51)